packages/storage/amplify_storage_s3_android/android/src/main/kotlin/com/amazonaws/amplify/amplify_storage_s3/types/FlutterPrefixResolver.kt

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazonaws.amplify.amplify_storage_s3.types
+
+import android.os.Handler
+import android.os.Looper
+import com.amplifyframework.core.Amplify
+import com.amplifyframework.core.Consumer
+import com.amplifyframework.storage.StorageAccessLevel
+import com.amplifyframework.storage.StorageException
+import com.amplifyframework.storage.s3.configuration.AWSS3PluginPrefixResolver
+import io.flutter.plugin.common.MethodChannel
+
+
+class FlutterPrefixResolver(methodChannel: MethodChannel) : AWSS3PluginPrefixResolver {
+
+    private var channel: MethodChannel = methodChannel
+
+    private val uiThreadHandler = Handler(Looper.getMainLooper())
+    private val LOG = Amplify.Logging.forNamespace("amplify:flutter:storage_s3")
+
+    override fun resolvePrefix(
+            accessLevel: StorageAccessLevel,
+            targetIdentity: String?,
+            onSuccess: Consumer<String>,
+            onError: Consumer<StorageException>
+    ) {
+
+        val accessLevelString = when (accessLevel) {
+            StorageAccessLevel.PUBLIC -> "public"
+            StorageAccessLevel.PROTECTED -> "protected"
+            StorageAccessLevel.PRIVATE -> "private"
```

Comment:
Should use `StorageAccessLevel` provided by amplify-android instead of hardcoding the string here.
[Example](https://github.com/aws-amplify/amplify-android/blob/main/aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/utils/S3Keys.java#L77)
```
StorageAccessLevel.PUBLIC.name().toLowerCase(Locale.US)
```